1-ECommerce/src/components/ProductCard.tsx

```diff
@@ -0,0 +1,43 @@
+import { FavoriteButton } from "./FavoriteButton";
+import type { Product } from "./ProductsContainer";
+
+interface ProductCardProp {
+  product: Product;
+  inFavorites: boolean;
+  onFavoriteClick: () => void;
+}
+
+export const ProductCard = ({
+  product,
+  inFavorites,
+  onFavoriteClick,
+}: ProductCardProp) => (
```

Comment:
Could the App benefit from wrapping this function in a call to `useMemo`? I'm not entirely sure that this component is "expensive" or "expensive enough" to warrant wrapping it in `useMemo`.

I'm also not sure if this will be of any benefit. Let's say a user clicks the "load more" button and another request is made, loading the same products that were previously loaded, and 6 new products. All these products replace the contents of the `products` variable, even though some of the products are "the same". Does React think that these products have changed? (If they have, `useMemo` is useless for my case because I would never use the cache and always run the `calculatedValue` function)

Reply:
From what I read: 
"React will compare each dependency with its previous value using the [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison."

And for comparing two objects: "`Object.is()` determines if two values (...) are the same object (meaning both values reference the same object in memory)"

Since I'm replacing the contents of the `products` array every time I make a call to an API, then the memory address of the prop is sure changing right? If so, maybe it's not useful to wrap this in `useMemo`, as I previously thought